launchlib/launcher.go

```diff
@@ -86,7 +86,33 @@ func compileCmdFromConfig(
 	var executable string
 	var executableErr error
 
-	if staticConfig.Type == "java" {
+	// Handle experimental nativeImage execution mode before standard processing
+	if customConfig.Experimental.ExecutionMode == ExecutionModeNative {
+		executable, executableErr = verifyPathIsSafeForExec(customConfig.Experimental.NativeImageExecutablePath)
+		if executableErr != nil {
+			return nil, executableErr
+		}
+		args = append(args, executable) // 0th argument is the command itself
+		// Add default system properties
+		args = append(args, "-Djava.io.tmpdir=var/data/tmp", "-Dsun.net.inetaddr.ttl=20")
+		for _, nativeArg := range customConfig.Experimental.NativeImageArguments {
+			args = append(args, nativeArg)
+		}
+		// If heapPercentage is specified, ensure MaxRAMPercentage flag is present
```

Comment:
nit: I'd say something like: 

// If `MaximumHeapSizePercent` is not present in `NativeImageArguments`, use `heapPercentage`.